### ISSUE TITLE
Jetty impl. of Websocket ServerEndpointConfig.Configurator lifecycle out of spec.

### ISSUE THREAD

**Issue by brettwooldridge** [eclipse/jetty.project#424]
The JavaDoc for the ``javax.websocket.server.ServerEndpointConfig.Configurator`` method ``modifyHandshake()`` states:

```
/**
 * Called by the container after it has formulated a handshake response resulting from
 * a well-formed handshake request. The container has already 
 * checked that this configuration has a matching URI, determined the 
 * validity of the origin using the checkOrigin method, and filled
 * out the negotiated subprotocols and extensions based on this configuration.
 * Custom configurations may override this method in order to inspect
 * the request parameters and modify the handshake response that the server has formulated.
 * and the URI checking also.
```
However, the [Jetty implementation](https://github.com/eclipse/jetty.project/blob/ac24196b0d341534793308d585161381d5bca4ac/jetty-websocket/javax-websocket-server-impl/src/main/java/org/eclipse/jetty/websocket/jsr356/server/JsrCreator.java#L89) is calling ``modifyHandshake()`` before ``checkOrigin()``, before determining the negotiated subprotocols, and before determining negotiated extensions.

This requires the following hack in order to get the negotiated protocols:
```java
@Override
public void modifyHandshake(ServerEndpointConfig config, HandshakeRequest request, HandshakeResponse response) {
   List<String> requested = ((JsrHandshakeRequest) request).getHeaders().get(HandshakeRequest.SEC_WEBSOCKET_PROTOCOL);

   String subprotocol = super.getNegotiatedSubprotocol(config.getSubprotocols(), requested);
   config.getUserProperties().put("subprotocol", subprotocol);
}
```

**Comment by joakime** [comment]
Corrected in jetty-9.2.x and merged up to jetty-9.3.x and master
The modifyHandshake and ServerEndpointConfig creation were improper, the rest was correct.

**Comment by brettwooldridge** [comment]
@joakime Thanks.
